rules/Carbon/Rector/New_/DateTimeInstanceToCarbonRector.php

```diff
@@ -56,22 +57,32 @@ public function getNodeTypes(): array
      */
     public function refactor(Node $node): ?Node
     {
-        if (! $this->isName($node->class, 'DateTime')) {
-            return null;
+        if ($this->isName($node->class, 'DateTime')) {
+            return $this->refactorWithClass($node, 'Carbon\\Carbon');
+        }
+
+        if ($this->isName($node->class, 'DateTimeImmutable')) {
+            return $this->refactorWithClass($node, 'Carbon\\CarbonImmutable');
         }
 
-        if ($node->isFirstClassCallable()) {
+        return null;
+    }
+
+    public function refactorWithClass(New_ $new, string $className): MethodCall|StaticCall|null
```

Comment:
I don't have the full picture probably but here I don't get why it is public (for who to call outside).

Else, thanks for the fix and test, I think next time, I'd be able to make a complete contribution at once 👌

Reply:
You're right, this should be private :+1: